blog/_posts/2018-04-19-scala-3.md

```diff
@@ -0,0 +1,105 @@
+ - The compiler can perform many of the rewritings automatically using a `-rewrite` option.
+ - Migration through automatic rewriting will also be offered through the [scalafix](https://github.com/scalacenter/scalafix) tool, which can convert sources to the cross-buildable language subset without requiring Scala 3 to be installed.
+
+_What’s the expected state of tool support?_
+
+ - Compiler: The Scala 3 compiler `dotc` has been used to compile itself and a growing set of libraries for a number of years now.
+ - IDEs: IDE support is provided by having `dotc` implement LSP, the Language Server Protocol,
+   including standard operations such as completion and hyperlinking and more advanced ones
+    such as find references or rename. There’s a [VS Code plugin](http://dotty.epfl.ch/docs/usage/ide-support.html) incorporating these operations.
+   JetBrains has also released a first version of Scala 3 support in their [Scala IntelliJ plugin](https://blog.jetbrains.com/scala/2017/03/23/scala-plugin-for-intellij-idea-2017-1-cleaner-ui-sbt-shell-repl-worksheet-akka-support-and-more), and we intend to work with them on further improvements.
+ - REPL: A friendly REPL is supported by the compiler
+ - Docs: A revamped Scaladoc tool generates docs for viewing in a browser and (in the future) also in the IDE..
+ - Build tools: There is a Dotty/Scala 3 plugin for sbt, and we will also work on Scala 3 integration in other build tools.
+
+_What about stability?_
+
+ - A [community build](https://github.com/lampepfl/dotty-community-build) contains some [initial open source projects](https://github.com/lampepfl/dotty-community-build/blob/master/src/test/scala/dotty/communitybuild/CommunityBuildTest.scala) that are compiled nightly using Scala 3. We plan to add a lot more projects to the build between now and the final release.
+ - We plan to use the period of developer previews to ensure that core projects are published for Scala 3.
+ - We have incorporated most of the Scala 2 regression tests in the Scala 3 test suite and will keep including new tests.
+ - In the near future we plan to build all Scala 3 tools using a previous version of the `dotc` compiler itself.
+   So far all tools are built first with the current Scala compiler and then again with `dotc`.
+   Basing the build exclusively on Scala 3 has the advantage that it lets us “eat our own dog food”
+   and try out the usability of Scala 3’s new language feature on a larger scale.
+
+_When can I try it out?_
+
+You can start working with Dotty now. See the getting started
+guide. Dotty releases are published every 6 weeks. We expect to be in feature-freeze
+and to release developer previews for Scala 3.0 in the first half of 2019.
+
+_What about macros?_
+
+Stay tuned! We are about to release another blog post specifically
+about that issue.
+
+_How can I help?_
+
+Scala 3 is developed completely in the open at
+[https://github.com/lampepfl/dotty]. Get involved there, by fixing and
```

Comment:
In order to make a URL a link in Markdown you need to use:

    [https://github.com/lampepfl/dotty](https://github.com/lampepfl/dotty)